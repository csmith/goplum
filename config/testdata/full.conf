defaults {
    alerts = ["text", "foo", "bar", ]
	good_threshold = 3
	interval = 1m30s
}

alert twilio.message "text" {
    sid = "xxx"
    token = "yyy"
    from = "+44 ..."
    to = "+44 ...."
}

check http.get "chameth.com" {
    url = "https://chameth.com/"
    content = "Personal Homepage of Chris Smith"
}

plugin http {
    foo = "bar"
}

group "webservices" {
    alert_limit = 3
    alert_window = 10m
    
    defaults {
        interval = 30s
        timeout = 10s
    }
}
