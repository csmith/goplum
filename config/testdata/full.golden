[
  {
    "Class": 17,
    "Line": 1,
    "Column": 0,
    "Value": "defaults"
  },
  {
    "Class": 2,
    "Line": 1,
    "Column": 9,
    "Value": 123
  },
  {
    "Class": 4,
    "Line": 2,
    "Column": 4,
    "Value": "alerts"
  },
  {
    "Class": 5,
    "Line": 2,
    "Column": 11,
    "Value": 61
  },
  {
    "Class": 14,
    "Line": 2,
    "Column": 13,
    "Value": 91
  },
  {
    "Class": 6,
    "Line": 2,
    "Column": 14,
    "Value": "text"
  },
  {
    "Class": 11,
    "Line": 2,
    "Column": 20,
    "Value": 44
  },
  {
    "Class": 6,
    "Line": 2,
    "Column": 22,
    "Value": "foo"
  },
  {
    "Class": 11,
    "Line": 2,
    "Column": 27,
    "Value": 44
  },
  {
    "Class": 6,
    "Line": 2,
    "Column": 29,
    "Value": "bar"
  },
  {
    "Class": 11,
    "Line": 2,
    "Column": 34,
    "Value": 44
  },
  {
    "Class": 15,
    "Line": 2,
    "Column": 36,
    "Value": 93
  },
  {
    "Class": 4,
    "Line": 3,
    "Column": 1,
    "Value": "good_threshold"
  },
  {
    "Class": 5,
    "Line": 3,
    "Column": 16,
    "Value": 61
  },
  {
    "Class": 7,
    "Line": 3,
    "Column": 18,
    "Value": 3
  },
  {
    "Class": 4,
    "Line": 4,
    "Column": 1,
    "Value": "interval"
  },
  {
    "Class": 5,
    "Line": 4,
    "Column": 10,
    "Value": 61
  },
  {
    "Class": 9,
    "Line": 4,
    "Column": 12,
    "Value": 90000000000
  },
  {
    "Class": 3,
    "Line": 5,
    "Column": 0,
    "Value": 125
  },
  {
    "Class": 16,
    "Line": 7,
    "Column": 0,
    "Value": "alert"
  },
  {
    "Class": 4,
    "Line": 7,
    "Column": 6,
    "Value": "twilio.message"
  },
  {
    "Class": 6,
    "Line": 7,
    "Column": 21,
    "Value": "text"
  },
  {
    "Class": 2,
    "Line": 7,
    "Column": 28,
    "Value": 123
  },
  {
    "Class": 4,
    "Line": 8,
    "Column": 4,
    "Value": "sid"
  },
  {
    "Class": 5,
    "Line": 8,
    "Column": 8,
    "Value": 61
  },
  {
    "Class": 6,
    "Line": 8,
    "Column": 10,
    "Value": "xxx"
  },
  {
    "Class": 4,
    "Line": 9,
    "Column": 4,
    "Value": "token"
  },
  {
    "Class": 5,
    "Line": 9,
    "Column": 10,
    "Value": 61
  },
  {
    "Class": 6,
    "Line": 9,
    "Column": 12,
    "Value": "yyy"
  },
  {
    "Class": 4,
    "Line": 10,
    "Column": 4,
    "Value": "from"
  },
  {
    "Class": 5,
    "Line": 10,
    "Column": 9,
    "Value": 61
  },
  {
    "Class": 6,
    "Line": 10,
    "Column": 11,
    "Value": "+44 ..."
  },
  {
    "Class": 4,
    "Line": 11,
    "Column": 4,
    "Value": "to"
  },
  {
    "Class": 5,
    "Line": 11,
    "Column": 7,
    "Value": 61
  },
  {
    "Class": 6,
    "Line": 11,
    "Column": 9,
    "Value": "+44 ...."
  },
  {
    "Class": 3,
    "Line": 12,
    "Column": 0,
    "Value": 125
  },
  {
    "Class": 18,
    "Line": 14,
    "Column": 0,
    "Value": "check"
  },
  {
    "Class": 4,
    "Line": 14,
    "Column": 6,
    "Value": "http.get"
  },
  {
    "Class": 6,
    "Line": 14,
    "Column": 15,
    "Value": "chameth.com"
  },
  {
    "Class": 2,
    "Line": 14,
    "Column": 29,
    "Value": 123
  },
  {
    "Class": 4,
    "Line": 15,
    "Column": 4,
    "Value": "url"
  },
  {
    "Class": 5,
    "Line": 15,
    "Column": 8,
    "Value": 61
  },
  {
    "Class": 6,
    "Line": 15,
    "Column": 10,
    "Value": "https://chameth.com/"
  },
  {
    "Class": 4,
    "Line": 16,
    "Column": 4,
    "Value": "content"
  },
  {
    "Class": 5,
    "Line": 16,
    "Column": 12,
    "Value": 61
  },
  {
    "Class": 6,
    "Line": 16,
    "Column": 14,
    "Value": "Personal Homepage of Chris Smith"
  },
  {
    "Class": 3,
    "Line": 17,
    "Column": 0,
    "Value": 125
  },
  {
    "Class": 19,
    "Line": 19,
    "Column": 0,
    "Value": "plugin"
  },
  {
    "Class": 4,
    "Line": 19,
    "Column": 7,
    "Value": "http"
  },
  {
    "Class": 2,
    "Line": 19,
    "Column": 12,
    "Value": 123
  },
  {
    "Class": 4,
    "Line": 20,
    "Column": 4,
    "Value": "foo"
  },
  {
    "Class": 5,
    "Line": 20,
    "Column": 8,
    "Value": 61
  },
  {
    "Class": 6,
    "Line": 20,
    "Column": 10,
    "Value": "bar"
  },
  {
    "Class": 3,
    "Line": 21,
    "Column": 0,
    "Value": 125
  },
  {
    "Class": 20,
    "Line": 23,
    "Column": 0,
    "Value": "group"
  },
  {
    "Class": 6,
    "Line": 23,
    "Column": 6,
    "Value": "webservices"
  },
  {
    "Class": 2,
    "Line": 23,
    "Column": 20,
    "Value": 123
  },
  {
    "Class": 4,
    "Line": 24,
    "Column": 4,
    "Value": "alert_limit"
  },
  {
    "Class": 5,
    "Line": 24,
    "Column": 16,
    "Value": 61
  },
  {
    "Class": 7,
    "Line": 24,
    "Column": 18,
    "Value": 3
  },
  {
    "Class": 4,
    "Line": 25,
    "Column": 4,
    "Value": "alert_window"
  },
  {
    "Class": 5,
    "Line": 25,
    "Column": 17,
    "Value": 61
  },
  {
    "Class": 9,
    "Line": 25,
    "Column": 19,
    "Value": 600000000000
  },
  {
    "Class": 17,
    "Line": 27,
    "Column": 4,
    "Value": "defaults"
  },
  {
    "Class": 2,
    "Line": 27,
    "Column": 13,
    "Value": 123
  },
  {
    "Class": 4,
    "Line": 28,
    "Column": 8,
    "Value": "interval"
  },
  {
    "Class": 5,
    "Line": 28,
    "Column": 17,
    "Value": 61
  },
  {
    "Class": 9,
    "Line": 28,
    "Column": 19,
    "Value": 30000000000
  },
  {
    "Class": 4,
    "Line": 29,
    "Column": 8,
    "Value": "timeout"
  },
  {
    "Class": 5,
    "Line": 29,
    "Column": 16,
    "Value": 61
  },
  {
    "Class": 9,
    "Line": 29,
    "Column": 18,
    "Value": 10000000000
  },
  {
    "Class": 3,
    "Line": 30,
    "Column": 4,
    "Value": 125
  },
  {
    "Class": 3,
    "Line": 31,
    "Column": 0,
    "Value": 125
  },
  {
    "Class": 1,
    "Line": 32,
    "Column": 0,
    "Value": null
  }
]